eldritch/test_events.py

```diff
@@ -1536,5 +1561,102 @@ def testInvalidSale(self):
       self.resolve_until_done()
 
 
+class CloseLocationTest(EventTest):
+  def testCloseForever(self):
+    place_name = "Woods"
+    self.state.event_stack.append(CloseLocation(place_name))
+    self.resolve_until_done()
+    self.assertTrue(self.state.places[place_name].closed)
+    self.char.place = self.state.places["Uptown"]
+    self.advance_turn(self.state.turn_number + 5, 'mythos')
+    self.resolve_until_done()
+    self.assertTrue(self.state.places[place_name].closed)
+
+  def testCloseWithGateForOneTurn(self):
+    place_name = "Woods"
+    self.advance_turn(self.state.turn_number, 'mythos')
+    self.resolve_until_done()
+    place = self.state.places[place_name]
+    place.gate = self.state.gates.popleft()
+    monster = next(iter(self.state.monsters))
+    monster.place = place
+    self.char.place = place
+    self.state.event_stack.append(CloseLocation(place_name, 1))
+    self.resolve_until_done()
+
+    self.assertEqual(place.closed_until, self.state.turn_number + 2)
+    self.assertEqual(self.char.place, place)
+    self.assertEqual(monster.place, place)
+    self.assertFalse(place.closed)
+
+    self.state.event_stack.append(GateCloseAttempt(self.char, place_name))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      fight_lore = self.resolve_to_choice(MultipleChoice)
+      fight_lore.resolve(self.state, fight_lore.choices[0])
+      self.resolve_until_done()
+
+    self.assertEqual(place.closed_until, self.state.turn_number + 2)
+    self.assertEqual(self.char.place.name, "Uptown")
+    self.assertEqual(monster.place.name, "Uptown")
+    self.assertTrue(place.closed)
+
+    self.advance_turn(self.state.turn_number + 1, 'movement')
+    movement = self.resolve_to_choice(CityMovement)
+    movement.resolve(self.state, movement.none_choice)
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
```

Comment:
Probably worth adding a comment here that you need to kill the cultist to continue advancing the turn properly.